test/staking/allocation.test.ts

```diff
@@ -484,193 +577,206 @@ describe('Staking:Allocation', () => {
     beforeEach(async function () {
       // Stake and allocate
       await staking.connect(indexer.signer).stake(tokensToStake)
-      await allocate(tokensToAllocate)
     })
 
-    it('reject close a non-existing allocation', async function () {
-      const invalidAllocationID = randomHexBytes(20)
-      const tx = staking.connect(indexer.signer).closeAllocation(invalidAllocationID, poi)
-      await expect(tx).revertedWith('!active')
-    })
-
-    it('reject close before at least one epoch has passed', async function () {
-      const tx = staking.connect(indexer.signer).closeAllocation(allocationID, poi)
-      await expect(tx).revertedWith('<epochs')
-    })
-
-    it('reject close if not the owner of allocation', async function () {
-      // Move at least one epoch to be able to close
-      await advanceToNextEpoch(epochManager)
+    for (const tokensToAllocate of [toBN(100), toBN(0)]) {
```

Comment:
nit: I see how this is practical to make the code shorter, but I wonder if it isn't cleaner to just put the two cases separately? Otherwise below you end up adding checks for whether the allocation is zero tokens or not, and it becomes a bit harder to follow what the test is doing.

Reply:
I find it was a bunch of code to copy/paste/repeat, let's think about how we can make them clearer in a different PR, maybe we can abstract part of it in different functions as behaviors.

Reply:
Yeah sounds good!